src/main.rs

```diff
@@ -155,7 +158,20 @@ async fn listen_blocks(
         .or_default_provider()
         .or_else(Region::new("eu-central-1"));
     let shared_config = aws_config::from_env().region(region_provider).load().await;
-    let client = Client::new(&shared_config);
+    let mut s3_conf = aws_sdk_s3::config::Builder::from(&shared_config);
+    // Owerride S3 endpoint in case you want to use custom solution
+    // like Minio or Localstack as a S3 compatible storage
+    if endpoint.is_some() {
```

Comment:
Any specific reason you're not using `if let Some(endpoint_string) = endpoint {}` to avoid `endpoint.as_ref().unwrap();`?

Reply:
I'm new to rust ecosystem and don't well know best practises yet. Thanks for pointing me on this.